cmd/common-main.go

```diff
@@ -169,10 +172,21 @@ func handleCommonEnvVars() {
 		globalIsEnvDomainName = true
 	}
 
-	globalDomainIP = os.Getenv("MINIO_PUBLIC_IP")
-	if globalDomainName != "" && globalDomainIP != "" && globalEtcdClient != nil {
+	minioEndpointsEnv, ok := os.LookupEnv("MINIO_PUBLIC_IPS")
+	if ok {
+		minioEndpoints := strings.Split(minioEndpointsEnv, ",")
+		globalDomainIPs = set.NewStringSet()
+		for _, ip := range minioEndpoints {
+			if !s3utils.IsValidIP(ip) {
+				err := errors.New("Invalid argument")
+				logger.FatalIf(err, "Unable to initialize Minio server with invalid MINIO_PUBLIC_IPS: %s", ip)
```

Comment:
since you are logging this per IP it is better to say one of the public IP or simply `MINIO_PUBLIC_IPS[i]: %s`